### PR TITLE
Added controller tests

### PR DESCRIPTION
**Pull request by goremukin**

### CODE REVIEW

File: src/app/Tests/AccountControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,614 @@
...
+            _userService.Verify(service => service.CreateUserAccountAsync(
+                It.Is<CreateUserGoogleModel>(m => m.Email == payload.Email &&  m.FirstName == payload.GivenName && m.LastName == payload.FamilyName)),
+                Times.Once
+            );
+        }
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async void SignInGoogleWithCodeShouldShouldEnableGoogleAuthWhenItDoesNotEnabled(bool isEnabled)
```
Comment (yarmolovichalex) [inline src/app/Tests/AccountControllerTests.cs:527]: The test is not passed:
`
Expected invocation on the mock should never have been performed, but was 1 times: service => service.CreateUserAccountAsync(It.IsAny<CreateUserGoogleModel>())
`

File: src/app/Api/Core/DAL/BaseRepository.cs [hunk 2]
```diff
@@ -51,7 +51,7 @@ public async Task InsertManyAsync(IEnumerable<TDocument> documents)
         public async Task<TDocument> FindOneAsync(TFilter filter)
         {
             var result = await Collection.FindAsync(BuildFilterQuery(filter));
-            return result.SingleOrDefault();
+            return result.FirstOrDefault();
```
Comment (yarmolovichalex) [inline src/app/Api/Core/DAL/BaseRepository.cs:54]: I'd keep `SingleOrDefault()`  for additional validation. This is different from [MongoDB behavior](https://docs.mongodb.com/manual/reference/method/db.collection.findOne/). But usually `FindOne` is used when we need to get a single unique item from DB (often by Id). If it gets more than one item, either the filter or the data may be broken and it's better to get aware of this as soon as possible.

File: src/app/Tests/AccountControllerTests.cs [hunk 3]
```diff
@@ -0,0 +1,614 @@
...
+                .ReturnsAsync(new User());
+
+            
+            var controller = CreateInstance();
+
+            // Act
+            var result = await controller.SignUpAsync(model);
+
+            // Assert
+            Assert.IsAssignableFrom<BadRequestResult>(result);
```
Comment (yarmolovichalex) [inline src/app/Tests/AccountControllerTests.cs:69]: Maybe, it's better to compare result type with assert type only - I mean, not to check if it's derived? I tested this assert, it also works:
```
Assert.IsType<BadRequestResult>(result);
```

File: src/app/Tests/AccountControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,614 @@
...
+
+            _appSettings = new AppSettings
+            {
+                WebUrl = "http://test.com",
+                LandingUrl = "http://test-landing.com"
+            };
+        }
+
+        [Fact]
+        public async void SignUpShouldReturnBadRequestWhenUserAlreadyExist()
```
Comment (yarmolovichalex) [inline src/app/Tests/AccountControllerTests.cs:51]: Could you confirm it's OK to return `void` there - not `Task`?

File: src/app/Tests/AccountControllerTests.cs (hunk 1 above)
Comment (goremukin) [reply src/app/Tests/AccountControllerTests.cs:527]: @yarmolovichalex, Will be passed after google oauth is fixed

**Pull request merged**